core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -1356,14 +1354,13 @@ public static String generateUniqueID() {
 			MessageDigest crypt = MessageDigest.getInstance("SHA-1");
 			crypt.reset();
 			crypt.update(id.getBytes());
-			uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
+			final String uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
```

Comment:
That one just seems to use `java.util.Random`, which would be vulnerable to being guessed by observing the generated ids.

UUID does use SecureRandom, but by concatenating a "less" random value, and generating a hash, you're potentially increasing the likeliness of collision.

I'll add a commit changing this to just use SecureRandom or UUID directly.